tests/unit_tests/db_engine_specs/test_drill.py

```diff
@@ -64,4 +65,27 @@ def test_sadrill_impersonation(app_context: AppContext) -> None:
     url = URL("drill+sadrill")
     username = "DoAsUser"
     DrillEngineSpec.modify_url_for_impersonation(url, True, username)
-    assert url.username == username
+    assert url.query["impersonation_target"] == username
+
+
+def test_invalid_impersonation(app_context: AppContext) -> None:
+    """
+    Test ``modify_url_for_impersonation`` method when driver == foobar.
+
+    The method raises an exception because impersonation is not supported
+    for drill+foobar.
+    """
+    from sqlalchemy.engine.url import URL
+
+    from superset.db_engine_specs.drill import DrillEngineSpec
+
+    url = URL("drill+foobar")
+    username = "DoAsUser"
+    exception_type = None
+
+    try:
+        DrillEngineSpec.modify_url_for_impersonation(url, True, username)
+    except Exception as e:
+        exception_type = type(e)
+
+    assert exception_type == SupersetDBAPIProgrammingError
```

Comment:
nit: this is how it's usually done with `pytest`:
```suggestion
    with pytest.raises(SupersetDBAPIProgrammingError):
        DrillEngineSpec.modify_url_for_impersonation(url, True, username)
```

Reply:
Oh - you may need to move the exception import into the test function